sympy/series/limits.py

```diff
@@ -273,16 +273,19 @@ def set_signs(expr):
             arg_flag = isinstance(expr, arg)
             sign_flag = isinstance(expr, sign)
             if abs_flag or sign_flag or arg_flag:
-                sig = limit(expr.args[0], z, z0, dir)
-                if sig.is_zero:
-                    sig = limit(1/expr.args[0], z, z0, dir)
-                if sig.is_extended_real:
-                    if (sig < 0) == True:
-                        return (-expr.args[0] if abs_flag else
-                                S.NegativeOne if sign_flag else S.Pi)
-                    elif (sig > 0) == True:
-                        return (expr.args[0] if abs_flag else
-                                S.One if sign_flag else S.Zero)
+                try:
+                    sig = limit(expr.args[0], z, z0, dir)
+                    if sig.is_zero:
+                        sig = limit(1/expr.args[0], z, z0, dir)
+                    if sig.is_extended_real:
+                        if (sig < 0) == True:
+                            return (-expr.args[0] if abs_flag else
+                                    S.NegativeOne if sign_flag else S.Pi)
+                        elif (sig > 0) == True:
+                            return (expr.args[0] if abs_flag else
+                                    S.One if sign_flag else S.Zero)
+                except NotImplementedError:
+                    pass
```

Comment:
The smallest amount of code possible should be inside the try/except block.

Why is this catching NotImplementedError anyway?

What is it that raises NotImplementedError and should that not be fixed instead?

Reply:
When computing limits like `limit(abs((-n/(n+1))**n), n ,oo)` this `set_signs` block computes 
`limit(((-n/(n+1))**n), n ,oo)` but unlike in master where the answer is given zero, we intend to raise an error for the same hence when computing the limit of the abs expression we can let it pass as gruntz will handle that downstream.

Reply:
@anutosh491 as we will be returning notimplementedd error for `limit(((-n/(n+1))**n), n ,oo)` then we need to catch it here other wise `limit(abs((-n/(n+1))**n), n ,oo)` wont work and will return error too.